### PR TITLE
Remove duplicated code and move duplicated strings to const

### PR DESCRIPTION
**Pull request by Rizwana777**
#### Description:
- Removed duplicate code and strings, moved duplicated strings to const in e2e test file

#### Link to JIRA Story (if applicable):
https://issues.redhat.com/browse/GITOPSRVCE-671

### CODE REVIEW

File: tests-e2e/argocd/argocd_instance_test.go [hunk 1]
```diff
@@ -87,23 +87,7 @@ var _ = Describe("Standalone ArgoCD instance E2E tests", func() {
...
-						Path:           "resources/test-data/sample-gitops-repository/environments/overlays/dev",
-						TargetRevision: "HEAD",
-					},
-					Destination: appv1.ApplicationDestination{
-						Name:      argocdv1.ClusterSecretName,
-						Namespace: destinationNamespace.Name,
-					},
-				},
-			}
+			app := appFixture.BuildArgoCDApplication("argo-app-01", argocdNamespace, fixture.RepoURL, fixture.GitopsDeploymentPath, "HEAD", "", "", argocdv1.ClusterSecretName, destinationNamespace.Name, &appv1.SyncPolicyAutomated{})
```
Comment (jgwest) [inline tests-e2e/argocd/argocd_instance_test.go:90]: As above, for the calls to BuildArgoCDApplication, IMHO it's a bit more elegant in this case to leave it as a Go obj.

File: tests-e2e/argocd/argocd_appproject_test.go [hunk 2]
```diff
@@ -69,26 +70,7 @@ var _ = Describe("ArgoCD AppProject E2E tests", func() {
...
-					Destination: appv1.ApplicationDestination{
-						Server:    "https://kubernetes.default.svc",
-						Namespace: fixture.GitOpsServiceE2ENamespace,
-					},
-					SyncPolicy: &appv1.SyncPolicy{
-						Automated: &appv1.SyncPolicyAutomated{},
-					},
-				},
-			}
+			app = appFixture.BuildArgoCDApplication("argo-app-01", dbutil.GetGitOpsEngineSingleInstanceNamespace(), "", fixture.GitopsDeploymentPath, "HEAD", appProject.Name, "https://kubernetes.default.svc", "", fixture.GitOpsServiceE2ENamespace, &appv1.SyncPolicyAutomated{})
```
Comment (jgwest) [inline tests-e2e/argocd/argocd_appproject_test.go:73]: This is totally just my opinion :grin: , but IMHO I think the original was a bit easier to read (the `BuildArgoCDApplication` parameter list is fairly long, and it's not necessarily obvious which field corresponds to what)

File: tests-e2e/argocd/queryscoped_cluster_secrets_test.go [hunk 3]
```diff
@@ -141,29 +141,10 @@ var _ = Describe("Argo CD Application tests", func() {
...
-							Destination: appv1alpha1.ApplicationDestination{
-								Name:      "test-" + userName,
-								Namespace: userName,
-							},
-							SyncPolicy: &appv1alpha1.SyncPolicy{
-								Automated: &appv1alpha1.SyncPolicyAutomated{Prune: true, SelfHeal: true},
-							},
-						},
-					}
+					argoCDApplication := application.BuildArgoCDApplication("test-"+userName, argoCDNamespace, "https://github.com/redhat-appstudio/managed-gitops/", fixture.GitopsDeploymentPath, "", "", "", "test-"+userName, userName, &appv1alpha1.SyncPolicyAutomated{Prune: true, SelfHeal: true})
```
Comment (jgwest) [inline tests-e2e/argocd/queryscoped_cluster_secrets_test.go:146]: As above, for the calls to BuildArgoCDApplication, IMHO it's a bit more elegant in this case to leave it as a Go obj.

File: tests-e2e/argocd/queryscoped_cluster_secrets_test.go [hunk 4]
```diff
@@ -215,26 +195,9 @@ var _ = Describe("Argo CD Application tests", func() {
...
-						Destination: appv1alpha1.ApplicationDestination{
-							Name:      "test-" + userName,
-							Namespace: otherUser, // Attempt to deploy into a Namespace we don't have access to
-						},
-						SyncPolicy: &appv1alpha1.SyncPolicy{
-							Automated: &appv1alpha1.SyncPolicyAutomated{Prune: true, SelfHeal: true},
-						},
-					},
-				}
+				// Attempt to deploy into a Namespace we don't have access to
```
Comment (jgwest) [inline tests-e2e/argocd/queryscoped_cluster_secrets_test.go:198]: As above, for the calls to BuildArgoCDApplication, IMHO it's a bit more elegant in this case to leave it as a Go obj.

**Review by jgwest — CHANGES_REQUESTED** [review]
Great work, this should make our E2E tests much easier to work with,  thanks @Rizwana777! A few more minor requests, then good to go.

**Comment by Rizwana777** [comment]
@jgwest Thank you for reviewing, addressed the comments, PTAl

**Review by jgwest — APPROVED** [review]
Looks great, thanks again @Rizwana777!

**Pull request merged**